massa-client/src/tests/scenarios.rs

```diff
@@ -1,40 +1,51 @@
+use anyhow::{bail, Result};
 use assert_cmd::Command;
 use serial_test::serial;
-use std::thread;
 
-#[tokio::test]
-#[serial]
-async fn test_if_exit_gracefully() {
-    let mut cmd = Command::cargo_bin("massa-client").unwrap();
-    cmd.arg("exit").assert().success();
+const TIMEOUT: u64 = 5;
+const TENTATIVES: u64 = 5;
+
+async fn run_client_cmd(cmd: &str) -> Result<String> {
+    for _ in 0..TENTATIVES {
+        let output = std::str::from_utf8(
+            &Command::new("cargo")
+                .args(&["run", "--", cmd, "--json"])
+                .assert()
+                .get_output()
+                .stdout,
+        )?
+        .to_string();
+        // TODO: compare with tcp connect error as a JSON object?
+        if !output.contains("tcp connect error") {
```

Comment:
This and the multiple tentative are only relevant for the first command after a node is launched, not for every command

Reply:
Humm, my concern was more: make the test robust to network issue/wait for the node to be started, if the node is already launched, this issue just not occur, so it not try several tentative/wait and if it fails for other reason it just propagates the error, this is for me the correct behavior.

Reply:
I would rather have a assert node is running function that retries a command until it's ok and then another one for just running a command and retrieving the output.